Source/JavaScriptCore/b3/B3Value.h

```diff
@@ -611,6 +611,10 @@ class JS_EXPORT_PRIVATE Value {
     template<typename... Arguments>
     void buildAdjacencyList(NumChildren numChildren, Arguments... arguments)
     {
+        size_t offset = computeAdjacencyListOffset();
+        RELEASE_ASSERT(offset == static_cast<uint16_t>(offset));
+        m_adjacencyListOffset = offset;
+
```

Comment:
Can you check whether this function is always called for all B3::Value creation?
Also, please check that this is correctly setup even when we refresh the current B3::Value in-place. (We use `new (this) Value` when we replace the node. See Value::replaceWith)

Reply:
Probably it is done correctly, but let's double-check :)

Reply:
Because we compute the offset in `buildAdjacencyList`, anywhere we are initializing the adjacency list should correctly set the offset. Specifically, we do this in every constructor of `Value` except the copy constructor. In the copy constructor's case though, it's a protected member of `Value` and only called from `Value::allocate` (which is what we call when cloning values). For `replaceWith`, we call a normal `Value` constructor, so we `buildAdjacencyList` and set the offset. So, we always set this field before we access the adjacency list of a value.